pkg/fieldpath/fieldpath.go

```diff
@@ -42,7 +42,20 @@ func ExtractFieldPathAsString(obj interface{}, fieldPath string) (string, error)
 		return "", nil
 	}
 
-	switch fieldPath {
+	path, subscript := SplitMaybeSubscriptedPath(fieldPath)
+
+	if len(subscript) > 0 {
+		switch path {
+		case "metadata.annotations":
+			return accessor.GetAnnotations()[subscript], nil
+		case "metadata.labels":
```

Comment:
I'd expect to validate the subscript is a valid labels key here

Reply:
Done.